Userland/Libraries/LibGfx/ImageFormats/TinyVGLoader.cpp

```diff
@@ -510,7 +510,27 @@ struct TinyVGLoadingContext {
 static ErrorOr<void> decode_header_and_update_context(TinyVGLoadingContext& context)
 {
     VERIFY(context.state == TinyVGLoadingContext::State::NotDecoded);
+
+    auto expected_color_table_size = [](u32 color_count, ColorEncoding color_encoding) -> u64 {
+        switch (color_encoding) {
+        case ColorEncoding::RGBA8888:
+            return color_count * 4;
+        case ColorEncoding::RGB565:
+            return color_count * 2;
+        case ColorEncoding::RGBAF32:
+            return color_count * 16;
+        case ColorEncoding::Custom:
+            break;
+        }
+
+        VERIFY_NOT_REACHED();
```

Comment:
This will now crash on `.tvg`s with a `Custom` encoding, rather returning the error from `decode_color_table()`.
I think this check should be moved into `decode_color_table()` (after the check for custom color encodings).

Reply:
:facepalm: I originally put the check into `decode_color_table()`, but moved it to try and avoid changing any function arguments. I've changed it back to what I did initially.

The fuzzer also spotted this immediately. I actually ran it before pushing this time!